src/libstd/macros.rs

```diff
@@ -305,10 +305,16 @@ macro_rules! eprintln {
 /// let _ = dbg!(a); // <-- `a` is moved again; error!
 /// ```
 ///
+/// You can also use `dbg!()` without a value to just print the
+/// file and line whenever it's reached.
+///
 /// [stderr]: https://en.wikipedia.org/wiki/Standard_streams#Standard_error_(stderr)
 #[macro_export]
 #[stable(feature = "dbg_macro", since = "1.32.0")]
 macro_rules! dbg {
+    () => {
+        eprintln!("[{}:{}]", file!(), line!());
```

Comment:
I would just let this call `dbg!(())` and make it a shorthand.

Reply:
@Centril the commit with `dbg!()` prints
```rust
[src/main.rs:2]
```

If it called `dbg!(())`, it would print
```rust
[src/main.rs:2] () = ()
```
Is that preferred?

Reply:
I feel like the noise would ruin the benefits of this. Separating out lines without the expression on them in the debug output with stuff like this might help make debugging visually easier. I think.

Reply:
Hmm; I mainly thought of the benefits as typing convenience, not changes in output...
Anyways, I don't mind this the way it is now -- we can always change it if we want since the format is unstable. (but please add a test for `let _: () = dbg!();` to make sure it compiles)

Reply:
I would prefer not to conflate “zero argument” with “one argument which has the unit type”, and printing `() = ()` seems noisy at best.